pyquil/api/quantum_computer.py

```diff
@@ -185,3 +189,165 @@ def wait_for_job(self, job_id):
+    either by a QVM simulator ("Quantum/Quil Virtual Machine") or a physical Rigetti QPU ("Quantum
+    Processing Unit") made of superconducting qubits.
+
+    You can choose the quantum computer to target through a combination of its name and optional
+    flags. There are multiple ways to get the same quantum computer. The following are equivalent::
+
+        >>> qc = get_qc("8Q-Agave-noisy-qvm")
+        >>> qc = get_qc("8Q-Agave", as_qvm=True, noisy=True)
+
+    and will construct a simulator of the 8q-agave chip with a noise model based on device
+    characteristics. We also provide a means for constructing generic quantum simulators that
+    are not related to a given piece of Rigetti hardware::
+
+        >>> qc = get_qc("9q-generic-qvm")
+        >>> qc = get_qc("9q-generic", as_qvm=True)
+
+    Redundant flags are acceptable, but conflicting flags will raise an exception::
+
+        >>> qc = get_qc("9q-generic-qvm") # qc is fully specified by its name
+        >>> qc = get_qc("9q-generic-qvm", as_qvm=True) # redundant, but ok
+        >>> qc = get_qc("9q-generic-qvm", as_qvm=False) # Error!
+
+    Use :py:func:`list_quantum_computers` to retrieve a list of known qc names.
+
+    This method is provided as a convenience to quickly construct and use QVM's and QPU's.
+    Power users may wish to have more control over the specification of a quantum computer
+    (e.g. custom noise models, bespoke topologies, etc.). This is possible by constructing
+    a :py:class:`QuantumComputer` object by hand. Please refer to the documentation on
+    :py:class:`QuantumComputer` for more information.
+
+    :param name: The name of the desired quantum computer. This should correspond to a name
+        returned by :py:func:`list_quantum_computers`. Names ending in "-qvm" will return
+        a QVM. Names ending in "-noisy-qvm" will return a QVM with a noise model. Otherwise,
+        we will return a QPU with the given name.
+    :param as_qvm: An optional flag to force construction of a QVM (instead of a QPU). If
+        specified and set to ``True``, a QVM-backed quantum computer will be returned regardless
+        of the name's suffix
+    :param noisy: An optional flag to force inclusion of a noise model. If
+        specified and set to ``True``, a quantum computer with a noise model will be returned
+        regardless of the name's suffix
+    :param connection: An optional :py:class:ForestConnection` object. If not specified,
+        the default values for URL endpoints, ping time, and status time will be used. Your
+        user id and API key will be read from ~/.pyquil_config. If you deign to change any
+        of these parameters, pass your own :py:class:`ForestConnection` object.
+    :return:
+    """
+    if connection is None:
+        connection = ForestConnection()
+
+    name, as_qvm, noisy = _parse_name(name, as_qvm, noisy)
+
+    if name == '9q-generic':
+        if not as_qvm:
```

Comment:
I thought that `as_qvm is None` still allows to use a qvm, whereas `as_qvm is False`. This condition,  however will evaluate `True` in both cases. Am I misunderstanding the parameter?

Reply:
I think I wanted to force the user to explicitly name that it's a qvm via `get_qc("9q-generic-qvm")` or `get_qc("9q-generic", as_qvm=True)`

It throws on `get_qc("9q-generic")` or `get_qc("9q-generic", as_qvm=False)`. Your question is about the former. Although we're unlikely to name any of our chips "generic", per naming convention that is more reminiscent of a qpu than a qvm.

This naming scheme puts real QPUs front and center (by having them lack suffixes), under the assumption that eventually people will be more interested in the qpu's than the associated qvms

Reply:
sounds good